### PR TITLE
Add online_image documentation

### PR DESCRIPTION
**Pull request by guillempages**
Add the documentation for the new feature "online image"

## Description:

**Related issue (if applicable):** related to esphome/feature-requests#1463

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4710

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by guillempages** [comment]
This is not stale, it is waiting on the code pr to be reviewed.

**Comment by clydebarrow** [comment]
Can I suggest the ability to specify an optional compile time image that acts as a placeholder until the downloaded image is available? I would assume this will need to be copied in `setup()`.

**Comment by nagyrobi** [comment]
Still only PNGs supported? A short note in the doc about supported formats would be useful.

A quick hint on the third party decoder libs LVGL supports on other environments: https://docs.lvgl.io/master/libs/index.html maybe some of them could match here too.

**Comment by guillempages** [comment]
@nagyrobi yes, for the moment only PNGs are supported. The code is "prepared" to add other formats (as long as the corresponding library supports "streaming decoding"), but the focus is on getting the basic functionality (i.e. PNG) in, before adding additional formats.

I did not document the image format on purpose, since the option "format: " is already implemented (but useless, since it only accepts PNG as a parameter), but I do see your point, that it would be nice to make the supported formats explicit in the documentation, and update it once a new format is added.

**Comment by guillempages** [comment]
> Can I suggest the ability to specify an optional compile time image that acts as a placeholder until the downloaded image is available? I would assume this will need to be copied in `setup()`.

As mentioned in the code-PR: in a follow-up PR; not in the initial one.

**Comment by nagyrobi** [comment]
> I did not document the image format on purpose

If you want this to be merged, at least _a note_ would be needed in order to avoid user confusion. They will assume any kind of image format is supported, just like in the offline image component.

**Comment by guillempages** [comment]
I've added a sentence in the overall description; I'll upgrade it to a note tomorrow.

**Comment by clydebarrow** [comment]
Two suggestions: consider moving this to its own page, rather than burying it in the Display section (there is other stuff in Display that needs the same, but no need to perpetuate that.)

Secondly, there needs to be more information on how to use it - how to actually trigger the download. Without extra config nothing happens because the `online_image` is a polling component but `update_interval` defaults to `never`.

File: components/display/index.rst [hunk 1]
```diff
@@ -733,6 +733,105 @@ be supplied to modify the color used to represent the on and off bits respective
...
+
+.. code-block:: yaml
+
+    display:
+      - platform: ...
+        id: my_display
+        # ...
+        lambda: |-
+          // Draw the image my_online_image at position [x=0,y=0]
+          it.image(0, 0, id(my_online_image))
```
Comment (clydebarrow) [inline components/display/index.rst]:
```suggestion
          it.image(0, 0, id(my_online_image));
```

Comment (guillempages) [reply components/display/index.rst]: Thanks for catching this

**Comment by guillempages** [comment]
I also thought, that the display component is having too many things inside, and graph, "image", "animation",... should be split out of that file, but thought I'd keep the new component consistent there.
I've moved it to a separate file now, and ~~might~~ created a separate PR to split those other components as well (#4101).

File: index.rst [hunk 2]
```diff
@@ -826,6 +826,8 @@ Display Components
     Display Menu Core, components/display_menu/index, folder-open.svg, dark-invert
     Graphical Display Menu, components/display_menu/graphical_display_menu, graphical_display_menu.png
     LCD Menu, components/display_menu/lcd_menu, lcd_menu.png
+    Font Renderer, components/display/fonts, format-font.svg, dark-invert
```
Comment (jesserockz) [inline index.rst]:
```suggestion
```

File: index.rst [hunk 3]
```diff
@@ -826,6 +826,7 @@ Display Components
     Display Menu Core, components/display_menu/index, folder-open.svg, dark-invert
     Graphical Display Menu, components/display_menu/graphical_display_menu, graphical_display_menu.png
     LCD Menu, components/display_menu/lcd_menu, lcd_menu.png
+    Online Image, components/online_image, image-sync-outline.svg, dark-invert
```
Comment (jesserockz) [inline index.rst]:
```suggestion
    Online Image, components/online_image, image-sync-outline.svg, dark-invert
    Display Menu Core, components/display_menu/index, folder-open.svg, dark-invert
    Graphical Display Menu, components/display_menu/graphical_display_menu, graphical_display_menu.png
    LCD Menu, components/display_menu/lcd_menu, lcd_menu.png
```

File: components/online_image.rst [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+.. warning::
+
+    This component requires a fair amount of RAM; both for downloading the image, and for storing the decoded image. It might work on devices without PSRAM, but there is no guarantee.
+
+This component has a dependency to :doc:`/components/http_request`; the configuration options you set to the ``http_request`` component will also apply here.
+
+.. code-block:: yaml
+
+    http_request:
+
```
Comment (jesserockz) [inline components/online_image.rst]: It's on the user to visit the above link to see how to configure the `http_request` component
```suggestion
```

File: components/online_image.rst [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+ - Add a ``component.update <image_id>`` in the ``on_connect:`` action on the :doc:`/components/wifi` component.
+ - Explicitly set an ``update_interval``.
+ - Call ``component.update <image_id>`` in an :doc:`/components/interval` block.
+ - Call ``component.update <image_id>`` where you need the image to be downloaded/updated.
+
+.. code-block:: yaml
+
+    wifi:
+      on_connect:
+        - component.update: my_online_image
```
Comment (jesserockz) [inline components/online_image.rst]: Minimal config example only first please. A full example is allowed to go below the `Actions` section

File: components/online_image.rst [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+  - ``BINARY``: Two colors, suitable for 1 color displays or 2 color image in color displays. Uses 1 bit
+    per pixel, 8 pixels per byte.
+  - ``TRANSPARENT_BINARY``: One color, any pixel that is fully transparent will not be drawn, and any other pixel
+    will be the on color. Uses 1 bit per pixel, 8 pixels per byte.
+  - ``GRAYSCALE``: Full scale grey. Uses 8 bits per pixel, 1 pixel per byte.
+  - ``RGB565``: Lossy RGB color stored. Uses 2 bytes per pixel.
+  - ``RGB24``: Full RGB color stored. Uses 3 bytes per pixel.
+  - ``RGBA``: Full RGB color stored. Uses 4 bytes per pixel. Any pixel with an alpha value < 127 will not be drawn.
+- **use_transparency** (*Optional*, boolean): If set the alpha channel of the input image will be taken into account, and pixels with alpha < 127 will not be drawn. For image types without explicit alpha channel, the color (0, 0, 1) (very dark blue) will be mapped to black, to be able to store transparency information within the image. Explicitly transparent types (``TRANSPARENT_BINARY`` and ``RGBA``) default to ``True`` and cannot be set to ``False``; other types default to ``False``.
```
Comment (jesserockz) [inline components/online_image.rst]:
```suggestion
- **use_transparency** (*Optional*, boolean): If set the alpha channel of the input image will be taken into account, 
  and pixels with alpha < 127 will not be drawn. For image types without explicit alpha channel, 
  the color (0, 0, 1) (very dark blue) will be mapped to black, to be able to store transparency information 
  within the image. Explicitly transparent types (``TRANSPARENT_BINARY`` and ``RGBA``) default to ``true`` and cannot be set to ``false``; other types default to ``false``.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**